src/scrcmd_c.c

```diff
@@ -3484,7 +3484,7 @@ BOOL ScrCmd_GiveLoanMon(ScriptContext *ctx) {
     u8 level = ScriptReadByte(ctx);
     u16 mapno = ScriptReadHalfword(ctx);
     HandleLoadOverlay(FS_OVERLAY_ID(npc_trade), OVY_LOAD_ASYNC);
-    NPCTrade_MakeAndGiveLoanMon(ctx->fieldSystem, tradeno, level, mapno);
+    NPCTrade_MakeAndGiveLoanMon(ctx->fieldSystem, (NpcTradeNum)tradeno, level, mapno);
```

Comment:
For these types of functions that always take in an NPC trade no arg make sure to put the enum type into the function declaration rather than casting to a 32 bit every time

Reply:
The function declarations already use the `NpcTradeNum` enum. Also it doesn't compile without the explicit cast to `NpcTradeNum`:

```
src\scrcmd_c.c:3487: illegal implicit conversion from 'unsigned char' to 'NpcTradeNum'
```